python-client/giskard/ml_worker/ml_worker.py

```diff
@@ -107,13 +103,12 @@ def _connect_websocket_client(self, is_server=False):
 
         try:
             if not is_server:
-                # External ML Worker: use jwt token
-                # raise ConnectFailedException
+                # External ML Worker: use API key
                 self.ws_conn.connect(
                     with_connect_command=True,
                     wait=False,
                     headers={
-                        "jwt": self.client.session.auth.token,
+                        "api-key": self.client.session.auth.token,
```

Comment:
Will we change auth for internal worker later?

Reply:
yes, internal worker will use persistent API key too, good point

Reply:
I think we should generate a "system" api key upon first Giskard boot or take it from env variable and use it by internal worker. The benefits compared to current jwt mechanism:
- system api key will remain the same between restarts
- depending on how we generate it it can be known by backend and ml worker at boot time (in case it's set in the env variable). This makes the auth mechanism easier as there's no need to wait for backend to boot to generate a jwt token

Reply:
No problem, it will be an internal change.
I created GSK-1663 and we can do it later, apart from this PR.